internal/pkg/manifest/svc.go

```diff
@@ -41,6 +41,47 @@ type ServiceImageWithPort struct {
 	Port         uint16 `yaml:"port"`
 }
 
+// Sidecar holds configuration for all sidecar containers in a service.
+type Sidecar struct {
+	Sidecars map[string]SidecarConfig `yaml:"sidecars"`
+}
+
+// SidecarConfig represents the configurable options for setting up a sidecar container.
+type SidecarConfig struct {
+	Port      string `yaml:"port"`
+	Image     string `yaml:"image"`
+	CredParam string `yaml:"credentialParameter"`
```

Comment:
```suggestion
	CredsParam string `yaml:"credentialsParameter"`
```
to be consistent with https://docs.aws.amazon.com/AmazonECS/latest/developerguide/private-auth.html

Reply:
`credentialsParameter` is consistent with the task def. However, in test def it is nested in `repositoryCredentials`, which I feel in Copilot we could prob simplify this unless in the future they add additional params under `repositoryCredentials`. What do you think?

Reply:
nvm i'll update to `credentialsParameter`